libres/tests/analysis/test_compare_initX.cpp

```diff
@@ -0,0 +1,181 @@
+#include <catch2/catch.hpp>
+#include <vector>
+#include <cmath>
+
+#include <ert/util/rng.h>
+#include <ert/enkf/enkf_util.hpp>
+
+#include <ert/util/bool_vector.hpp>
+#include <ert/enkf/obs_data.hpp>
+#include <ert/enkf/meas_data.hpp>
+#include <ert/analysis/ies/ies_enkf.hpp>
+#include <ert/analysis/std_enkf.hpp>
+
+struct model {
+    double amplitude;
+    double phase;
+
+    model(double a, double p) : amplitude(a), phase(p) {}
+
+    model(rng_type *rng) {
+        this->amplitude = 0.25 + enkf_util_rand_normal(1, 0.5, rng);
+        this->phase = enkf_util_rand_normal(0, M_PI / 4, rng);
+    }
+
+    double eval(double x) const {
+        return this->amplitude * std::sin(x + this->phase);
+    }
+};
+
+TEST_CASE("compare_initX", "[analysis]") {
+    const char *obs_key = "OBS1";
+    const int ens_size = 100;
+    const int obs_size = 10;
+    std::vector<model> ens;
+    bool_vector_type *ens_mask = bool_vector_alloc(ens_size, true);
+    obs_data_type *obs_data = obs_data_alloc(1.0);
+    meas_data_type *meas_data = meas_data_alloc(ens_mask);
+    rng_type *rng = rng_alloc(MZRAN, INIT_DEFAULT);
+    model true_model{1.0, 0.0};
+
+    for (int iens = 0; iens < ens_size; iens++)
+        ens.push_back(model(rng));
+
+    meas_block_type *mb = meas_data_add_block(meas_data, obs_key, 1, obs_size);
+    obs_block_type *ob =
```

Comment:
What is `ob` used for?
I see that we set some values on line 55, but I don't understand how it's used.

Reply:
Well - we need that handle to the interior of the obs_data in order to det the observer values

Reply:
Ah, so doing `obs_block_iset(ob, iobs, true_model.eval(x), 0.25);` updates `obs_data` which is later used to initialize matrices?
My Python-brain hurts.

Reply:
Yes that is correct; you should go to an old School C shrink!

Reply:
Most definitely!